### PR TITLE
fix: fix test connection in teambition

### LINKED ISSUE #4890: [Bug][teambition] test connection not work
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

test connection fail on list page
image

test connection success even though auth is wrong
image

### What do you expect to happen

1. test connection success on list page
2. test connection fail when auth is wrong

### How to reproduce

1. create a teambition connection
2. test connection

### Anything else

_No response_

### Version

main

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by likyh**
### Summary

1. fix test connection in teambition list page

image

2. fix test when appid is wrong
image

### Does this close any open issues?
Closes #4890

### CODE REVIEW

**Comment by likyh** [comment]
@coldgust Hi, I fix a bug in teambition. Need your review.

**Comment by coldgust** [comment]
LGTM

File: backend/plugins/teambition/api/connection.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func TestConnection(input *plugin.ApiResourceInput) (*plugin.ApiResourceOutput,
 		return nil, err
 	}
 	if resBody.Code != http.StatusOK {
-		return nil, errors.HttpStatus(res.StatusCode).New(fmt.Sprintf("unexpected status code: %d", res.StatusCode))
+		return nil, errors.HttpStatus(resBody.Code).New(fmt.Sprintf("unexpected code: %d, %s", resBody.Code, resBody.ErrorMessage))
```
Comment (klesh) [inline backend/plugins/teambition/api/connection.go:73]: Can you provide the doc link to support that `resBody.Code` is in fact a `HttpStatusCode`?

Comment (likyh) [reply backend/plugins/teambition/api/connection.go:73]: Emm, no doc found

**Comment by likyh** [comment]
https://github.com/apache/incubator-devlake/pull/5066

**Pull request closed without merging**